.github/deployment/scripts/protocol-cli.rs

```diff
@@ -681,56 +742,63 @@ async fn whitelist_actions(addresses_file: &str, private_key: Option<&str>) -> R
         }
     }
 
-    info(&format!("✓ Action whitelisting completed! Processed {} contracts", action_contracts.len()));
+    info(&format!("✓ Completed whitelisting {} contracts for {}", action_contracts.len(), chain));
     Ok(())
 }
 
 // =============================================================================
 // UPDATE CHAIN CONFIG
 // =============================================================================
 
-/// Update chain configuration with deployed contract addresses for a specific network
-async fn update_chain_config(addresses_file: &str, chain_config_file: &str, network: &str) -> Result<()> {
-    let addresses: HashMap<String, String> = serde_json::from_str(&fs::read_to_string(addresses_file)?)?;
+/// Update chain configuration with deployed contract addresses
+async fn update_chain_config(config: &DeploymentConfig, addresses_file: &str, chain_config_file: &str) -> Result<()> {
+    let addresses_by_chain: HashMap<String, HashMap<String, String>> =
+        serde_json::from_str(&fs::read_to_string(addresses_file)?)?;
     let mut chain_config: serde_json::Value = serde_json::from_str(&fs::read_to_string(chain_config_file)?)?;
+    let chains = config.chains.as_ref().filter(|c| !c.is_empty())
+        .ok_or_else(|| anyhow!("No chains in config"))?;
+    let tier_mapping = get_contract_mapping();
+    let mut has_changes = false;
 
-    let target_key = chain_config.as_object()
-        .and_then(|obj| obj.iter().find(|(_, v)| v.get("name").and_then(|n| n.as_str()) == Some(network)))
-        .map(|(k, _)| k.clone())
-        .ok_or_else(|| anyhow!("Network '{}' not found", network))?;
+    for chain in chains {
+        let chain_id = get_chain_info(chain).ok_or_else(|| anyhow!("Unknown chain: {}", chain))?.chain_id;
+        let addresses = addresses_by_chain.get(chain)
+            .ok_or_else(|| anyhow!("No addresses found for '{}'", chain))?;
 
-    let target_block = chain_config[&target_key].as_object_mut()
-        .ok_or_else(|| anyhow!("Invalid network block"))?;
+        let target_key = chain_config.as_object()
+            .and_then(|obj| obj.iter().find(|(k, _)| k.parse::<u64>().ok() == Some(chain_id)))
+            .map(|(k, _)| k.clone())
+            .ok_or_else(|| anyhow!("Chain '{}' (chain_id: {}) not found in config", chain, chain_id))?;
 
-    let tier_mapping = get_contract_mapping();
-    let mut changes = false;
-    for (contract_name, address) in &addresses {
-        if let Some(details) = tier_mapping.values().find_map(|tier| tier.contracts.get(contract_name)) {
-            if let Some(key) = &details.chain_config_key {
+        let target = chain_config[&target_key].as_object_mut().ok_or_else(|| anyhow!("Invalid chain block"))?;
+        let mut updated = false;
+
+        for (contract_name, address) in addresses {
+            if let Some(key) = tier_mapping.values().find_map(|t| t.contracts.get(contract_name)?.chain_config_key.as_ref()) {
```

Comment:
you could use a different pattern here to avoid the streetfighter code, check https://doc.rust-lang.org/rust-by-example/error/option_unwrap/and_then.html